File: models/Gpt-3.5-turbo/cpp/code/problems/mathematics/Common_Divisors.cpp

#include <iostream>
#include <algorithm>

int gcd(int a, int b) {
    if (b == 0) {
        return a;
    }
    return gcd(b, a % b);
}

int main() {
    int n;
    std::cin >> n;

    int arr[n];
    for (int i = 0; i < n; i++) {
        std::cin >> arr[i];
    }

    int maxGcd = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            int currentGcd = gcd(arr[i], arr[j]);
            maxGcd = std::max(maxGcd, currentGcd);
        }
    }

    std::cout << maxGcd << std::endl;

    return 0;
}

File: models/Gpt-3.5-turbo/py/code/mathematics/Common_Divisors.py

from math import gcd

def find_max_gcd(n, arr):
    max_gcd = 0
    
    for i in range(n):
        for j in range(i+1, n):
            max_gcd = max(max_gcd, gcd(arr[i], arr[j]))
    
    return max_gcd

# Read input
n = int(input())
arr = list(map(int, input().split()))

# Find maximum GCD
max_gcd = find_max_gcd(n, arr)

# Print output
print(max_gcd)
